R/classify.R

```diff
@@ -0,0 +1,57 @@
+# Function to calculate mode
+calculate_mode <- function(x) {
+  ux <- unique(x)
+  ux[which.max(tabulate(match(x, ux)))]
+}
+
+# Function to calculate mode behavior for each window of N minutes
+plot_mode_behavior_by_window <- function(data,
+                                         time_col,
+                                         behavior_col,
+                                         window_minutes) {
+  data %>%
+    dplyr::mutate(time = ymd_hms(!!sym(time_col)),
+                  # Ensure the time column is in POSIXct format
+                  window = floor_date(time, unit = paste0(window_minutes, " mins"))) %>%
+    dplyr::group_by(window) %>%
+    dplyr::summarise(mode_behavior = calculate_mode(!!sym(behavior_col))) %>%
+    dplyr::filter(!is.na(mode_behavior)) -> mode_sum
+  mode_sum %>%
+    ggplot2::ggplot(ggplot2::aes(x = window, y = mode_behavior, color = mode_behavior)) +
+    ggplot2::geom_point(size = 3) +
+    ggplot2::labs(
+      x = "Time",
+      y = "Mode Behavior",
+      title = paste0("Mode Behavior Every ", window_minutes, " Minutes")
+    ) +
+    ggplot2::theme_minimal() -> plotting
+  print(plotting)
+  return(mode_sum)
+}
+
+#' classify_behaviors
+#'
+#' This function takes a data frame and a classifier and predicts across the data frame assuming everything lines up
+#'
+#' @param dat A data frame containing the data.
+#' @param MSOM_path classifier
+#' @param quiet chill out
+#' @return A data frame with a column of behaviors
+#' @rdname classify_behaviors
+#' @export
+#'
+classify_behaviors <- function(
+                               dat,
+                               MSOM_path = "tests/testthat/MSOM_8by7.rda",
```

Comment:
This will break when used in package. So better make it available as package data.

Reply:
But also, see comment above about file size

Reply:
> make it available as package data

I forgot how to do this

Reply:
i think i once knew, but can't remember

Reply:
there is something in the object called "whatmap" that related to the data and looks like it's necessary...not simple...

Reply:
I have a solution to filesize, will push later today or tomorrow. Off to EERC workshop